### ISSUE TITLE
Support per-component strides in the picture abstraction

### ISSUE THREAD

**Issue by lu-zero** [ultravideo/kvazaar#98]
The change is somehow invasive but conceptually simple and should be needed to support other pixel formats.

**Comment by Venti-** [comment]
Yeah sure, seems like we have to do that eventually. Is this also to avoid having to copy input pixels from libav buffers to Kvazaar internal ones?

**Comment by lu-zero** [comment]
I reviewed that part, to make that part working you'd need to expose also the destructor callback and a private pointer to pass to it.

This way it can be zero-copied and would also lead the way to have it support opaque buffers assuming the opencl effort would let them use them directly as now we almost have for QSV (in that case we have decoders natively outputting opaque buffers and encoders consuming exactly the same kind)

**Comment by Venti-** [comment]
Ok, good to know. We will look into it.
